framework/caching/Cache.php

```diff
@@ -537,4 +538,44 @@ public function offsetUnset($key)
     {
         $this->delete($key);
     }
+
+    /**
+     * Method combines both [[set()]] and [[get()]] methods to retrieve value identified by a $key,
+     * or to store the result of $closure execution if there is no cache available for the $key.
+     *
+     * Usage example:
+     *
+     * ```php
+     * public function getTopProducts($count = 10) {
+     *     $cache = $this->cache; // Could be Yii::$app->cache
+     *     return $cache->closure(['top-n-products', 'n' => $count], function ($cache) use ($count) {
+     *         return Products::find()->mostPopular()->limit(10)->all();
+     *     }, 1000);
+     * }
+     * ```
+     *
+     * @param mixed $key a key identifying the value to be cached. This can be a simple string or
+     * a complex data structure consisting of factors representing the key.
+     * @param \Closure $closure the closure that will be used to generate a value to be cached
```

Comment:
@SilverFire I think you should add a note like: "Note, that the return value `false` will not get cached.".